admin/listPurchases/listView.html

```diff
@@ -93,6 +93,13 @@
     }
     tbody.innerHTML = renderEntries(entries);
 
+    Array.prototype.forEach.call(document.querySelectorAll('.mark-as-sent-btn'), function(markAsSentButton) {
+        markAsSentButton.addEventListener('click', (e) => {
+            e.preventDefault();
+            markOrderDispatched(markAsSentButton.dataset.orderId);
```

Comment:
The first lines in `markOrderDispatched` are:

```js
    let dispatchButton = document.querySelector(".more-info-" + id + " .mark-as-sent-btn");
    dispatchButton.classList.add("is-clicked");
```

Basically you have the `markAsSentButton` in scope here, but pass only its `orderId` into the function. Inside the function, the first thing you do is find that button again by its ID. It would be much faster to pass in a button reference, but that makes the code confusing becuase `markOrderDispatched` doesn't seem like it should take a button as an argument.

I suggested further below to rename the function. The alternative is to move the button handling code out here, e.g.:

```js
    Array.prototype.forEach.call(document.querySelectorAll('.mark-as-sent-btn'), function(markAsSentButton) {
        markAsSentButton.addEventListener('click', (e) => {
            e.preventDefault();
            markAsSentButton.classList.add("is-clicked");
            markOrderDispatched(markAsSentButton.dataset.orderId);
        });
    });
```

This removes the need for the first two lines inside the functino.